runs/repository/interfaces/common.go

```diff
@@ -0,0 +1,30 @@
+package interfaces
+
+// ListResourceInput contains parameters for querying collections of resources.
+type ListResourceInput struct {
+	Limit  int
+	Offset int
+
+	Filter Filter
+	// The filter set by scopeBy in the query
+	ScopeByFilter  Filter
+	SortParameters []SortParameter
+}
+
+func (l ListResourceInput) WithFilter(filter Filter) ListResourceInput {
+	if l.Filter != nil {
+		l.Filter = l.Filter.And(filter)
+	}
+
+	l.Filter = filter
```

Comment:
Please wrap l.Filter = filter into an else block, otherwise this function will overwrite all existing filters with input argument
```suggestion
	} else {
	  l.Filter = filter
	}
```